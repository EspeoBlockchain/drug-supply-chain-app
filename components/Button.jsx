import React from 'react';
import {
  View,
  StyleSheet,
  TouchableOpacity,
  Text,
} from 'react-native';
import PropTypes from 'prop-types';
import layout from '../constants/Layout';

const styles = StyleSheet.create(layout.styles);

const button = (props) => {
  const { disabled, onButtonPress, text } = props;

  return (
    <View style={styles.buttonContainer}>
      <TouchableOpacity style={styles.button} onPress={onButtonPress} disabled={disabled}>
        <Text style={styles.buttonText}>{text}</Text>
      </TouchableOpacity>
    </View>
  );
};

button.propTypes = {
  disabled: PropTypes.bool,
  onButtonPress: PropTypes.func,
  text: PropTypes.string,
};

button.defaultProps = {
  disabled: false,
  onButtonPress: () => null,
  text: '',
};

export default button;
