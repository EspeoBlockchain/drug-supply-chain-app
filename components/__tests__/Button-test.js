import 'react-native';
import React from 'react';
import Button from '../Button';
import renderer from 'react-test-renderer';

it('renders corectly', () => {
    const tree = renderer.create(<Button />).toJSON();

    expect(tree).toMatchSnapshot();
});
