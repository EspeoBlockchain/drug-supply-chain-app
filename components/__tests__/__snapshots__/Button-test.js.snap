// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders corectly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "paddingVertical": 50,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "borderRadius": 15,
        "color": "#d9572c",
        "marginHorizontal": 20,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#d9572c",
          "fontFamily": "raleway",
          "fontSize": 20,
          "padding": 20,
        }
      }
    >
      
    </Text>
  </View>
</View>
`;
